src/ATen/native/xpu/sycl/fused_adam_impl.cpp

```diff
@@ -0,0 +1,105 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+
+#include <ATen/native/xpu/sycl/MultiTensorApply.h>
+#include <ATen/native/xpu/sycl/fused_adam_utils.h>
+
+namespace at::native::xpu {
+
+void _fused_adam_xpu_impl_(
+    at::TensorList params,
+    at::TensorList grads,
+    at::TensorList exp_avgs,
+    at::TensorList exp_avg_sqs,
+    at::TensorList state_steps,
+    const double lr,
+    const double beta1,
+    const double beta2,
+    const double weight_decay,
+    const double eps,
+    const bool maximize,
+    const c10::optional<at::Tensor>& grad_scale,
+    const c10::optional<at::Tensor>& found_inf) {
+  std::vector<std::vector<at::Tensor>> tensor_lists{
+      params.vec(), grads.vec(), exp_avgs.vec(), exp_avg_sqs.vec()};
+
+  if (grad_scale.has_value())
+    TORCH_WARN(" grad_scale not supported with ipex fused_adam.")
```

Comment:
Why we have such limitation ?

Reply:
optional parameters (grad_scale, has_inf), are now supported and aligned to CUDA implementation